docs: Changes to settings docs for v1 (#6677)
aws/karpenter-provider-aws: website/content/en/preview/reference/settings.md

| KARPENTER_SERVICE | \-\-karpenter-service | The Karpenter Service name for the dynamic webhook certificate|
| KUBE_CLIENT_BURST | \-\-kube-client-burst | The maximum allowed burst of queries to the kube-apiserver (default = 300)|
| KUBE_CLIENT_QPS | \-\-kube-client-qps | The smoothed rate of qps to kube-apiserver (default = 200)|

> | DISABLE_LEADER_ELECTION | \-\-disable-leader-election | Start leader election client and gain leadership before executing the main loop. Enable this when running replicated components for high availability.|

Reviewer: ```suggestion
| DISABLE_LEADER_ELECTION | \-\-disable-leader-election | Disable the leader election client. Disable this when running a single replica components. (default = false)|
```

Reviewer 2: Did we run `make docgen` on this? It should automatically remove these and update their descriptions IIRC

| LOG_LEVEL | \-\-log-level | Log verbosity level. Can be one of 'debug', 'info', or 'error' (default = info)|
| LOG_OUTPUT_PATHS | \-\-log-output-paths | Optional comma separated paths for directing log output (default = stdout)|
| LOG_ERROR_OUTPUT_PATHS | \-\-log-error-output-paths | Optional comma separated paths for directing log error output (default = stderr)|
| MEMORY_LIMIT | \-\-memory-limit | Memory limit on the container running the controller. The GC soft memory limit is set to 90% of this value. (default = -1)|
| METRICS_PORT | \-\-metrics-port | The port the metric endpoint binds to for operating metrics about the controller itself (default = 8080)|
| RESERVED_ENIS | \-\-reserved-enis | Reserved ENIs are not included in the calculations for max-pods or kube-reserved. This is most often used in the VPC CNI custom networking setup https://docs.aws.amazon.com/eks/latest/userguide/cni-custom-network.html. (default = 0)|
| VM_MEMORY_OVERHEAD_PERCENT | \-\-vm-memory-overhead-percent | The VM memory overhead as a percent that will be subtracted from the total memory for all instance types. (default = 0.075)|

> | WEBHOOK_METRICS_PORT | \-\-webhook-metrics-port | The port the webhook metric endpoing binds to for operating metrics about the webhook (default = 8081)|

Reviewer: We ended up not making this change. I think there's a few more things we'll be doing but then we'll run `make docgen` again.
